include/data/field.h

```diff
@@ -36,6 +36,7 @@ struct FieldHighType {
     STRING,
     POINTER,
     ENUM,
+    MODE_GUARD,
```

Comment:
Maybe it would be better to always use the same name for guard? (hopefully) you always use enum with qualified name, so it's unambigous anyway.

Also it would be good for @koriakin to review adding those.